src/mgr/PyModules.cc

```diff
@@ -544,8 +544,11 @@ void PyModules::set_config(const std::string &handle,
   }
   set_cmd.wait();
 
-  // FIXME: is config-key put ever allowed to fail?
-  assert(set_cmd.r == 0);
+  // config-key put will fail if mgr's auth key has insufficient
+  // permission to set config keys
+  // FIXME: should this somehow raise an exception back into Python land?
+  dout(0) << "`config-key put " << global_key << " " << val << "` failed: "
```

Comment:
if `set_cmd.r` is 0, shall we continue happily without printing this error message?